external/storm-hdfs/src/main/java/org/apache/storm/hdfs/spout/HdfsSpout.java

```diff
@@ -0,0 +1,739 @@
+    }
+
+
+    if ( conf.containsKey(configKey) ) {
+      Map<String, Object> map = (Map<String, Object>)conf.get(configKey);
+        if(map != null) {
+          for(String keyName : map.keySet()){
+            LOG.info("HDFS Config override : " + keyName + " = " + String.valueOf(map.get(keyName)));
+            this.hdfsConfig.set(keyName, String.valueOf(map.get(keyName)));
+          }
+          try {
+            HdfsSecurityUtil.login(conf, hdfsConfig);
+          } catch (IOException e) {
+            LOG.error("HDFS Login failed ", e);
+            throw new RuntimeException(e);
+          }
+        } // if(map != null)
+      }
+
+    // Reader type config
+    if( conf.containsKey(Configs.READER_TYPE) ) {
+      readerType = conf.get(Configs.READER_TYPE).toString();
+      checkValidReader(readerType);
+    }
+
+    // -- source dir config
+    if ( !conf.containsKey(Configs.SOURCE_DIR) ) {
+      LOG.error(Configs.SOURCE_DIR + " setting is required");
+      throw new RuntimeException(Configs.SOURCE_DIR + " setting is required");
+    }
+    this.sourceDirPath = new Path( conf.get(Configs.SOURCE_DIR).toString() );
+
+    // -- archive dir config
+    if ( !conf.containsKey(Configs.ARCHIVE_DIR) ) {
+      LOG.error(Configs.ARCHIVE_DIR + " setting is required");
+      throw new RuntimeException(Configs.ARCHIVE_DIR + " setting is required");
+    }
+    this.archiveDirPath = new Path( conf.get(Configs.ARCHIVE_DIR).toString() );
+    validateOrMakeDir(hdfs, archiveDirPath, "Archive");
+
+    // -- bad files dir config
+    if ( !conf.containsKey(Configs.BAD_DIR) ) {
+      LOG.error(Configs.BAD_DIR + " setting is required");
+      throw new RuntimeException(Configs.BAD_DIR + " setting is required");
+    }
+
+    this.badFilesDirPath = new Path(conf.get(Configs.BAD_DIR).toString());
+    validateOrMakeDir(hdfs, badFilesDirPath, "bad files");
+
+    // -- ignore file names config
+    if ( conf.containsKey(Configs.IGNORE_SUFFIX) ) {
+      this.ignoreSuffix = conf.get(Configs.IGNORE_SUFFIX).toString();
+    }
+
+    // -- lock dir config
+    String lockDir = !conf.containsKey(Configs.LOCK_DIR) ? getDefaultLockDir(sourceDirPath) : conf.get(Configs.LOCK_DIR).toString() ;
+    this.lockDirPath = new Path(lockDir);
+    validateOrMakeDir(hdfs,lockDirPath,"locks");
+
+    // -- lock timeout
+    if( conf.get(Configs.LOCK_TIMEOUT) !=null ) {
+      this.lockTimeoutSec = Integer.parseInt(conf.get(Configs.LOCK_TIMEOUT).toString());
+    }
+
+    // -- enable/disable ACKing
+    Object ackers = conf.get(Config.TOPOLOGY_ACKER_EXECUTORS);
+    if( ackers!=null ) {
+      int ackerCount = Integer.parseInt(ackers.toString());
+      this.ackEnabled = (ackerCount>0);
+      LOG.debug("ACKer count = {}", ackerCount);
+    }
+    else {
+      this.ackEnabled = false;
+      LOG.debug("No ACKers config found");
+    }
+
+    LOG.info("ACK mode is {}", ackEnabled ? "enabled" : "disabled");
```

Comment:
Again.  If acking is disabled ack is called on the spout as part of the emit.  You should not need all of this extra logic.  Storm should handle all of that for you.